lib/alloc.js

```diff
@@ -0,0 +1,43 @@
+class CodeAllocator {
+  constructor (sliceSize) {
+    this.sliceSize = sliceSize;
+    this.slicesPerPage = Process.pageSize / sliceSize;
+
+    this.pages = [];
+    this.free = [];
+  }
+
+  allocateSlice () {
+    const slice = this.free.pop();
+    if (slice !== undefined) {
+      return slice;
+    }
+
+    return this._allocatePage();
+  }
+
+  _allocatePage () {
+    const page = Memory.alloc(Process.pageSize);
+
+    const { sliceSize, slicesPerPage } = this;
+
+    for (let i = 1; i !== slicesPerPage; i++) {
```

Comment:
Is there a reason we start at offset 1 here instead of 0? In Gum's code allocator we use the 0th slice to store metadata, but here we store that on the JS side.

Reply:
Sorry, missed your earlier comment. Please disregard this :)